tests/pytests/test_issues.py

```diff
@@ -867,17 +867,20 @@ def mod5778_add_new_shard_to_cluster(env: Env):
 
     # Move one slot (0) to the new shard (according to https://redis.io/commands/cluster-setslot/)
     new_shard_id = new_shard_conn.execute_command('CLUSTER MYID')
+    source_shard_id = conn.execute_command('CLUSTER MYID')
+    env.assertEqual(new_shard_conn.execute_command(f"CLUSTER SETSLOT 0 IMPORTING {source_shard_id}"), "OK")
+    env.assertEqual(conn.execute_command(f"CLUSTER SETSLOT 0 MIGRATING {new_shard_id}"), "OK")
     env.assertEqual(new_shard_conn.execute_command(f"CLUSTER SETSLOT 0 NODE {new_shard_id}"), "OK")
     env.assertEqual(conn.execute_command(f"CLUSTER SETSLOT 0 NODE {new_shard_id}"), "OK")
 
     # Now we expect that the new shard will be a part of the cluster partition in redisearch (allow some time
     # for the cluster refresh to occur and acknowledged by all shards)
-    while True:
-        time.sleep(0.5)
-        cluster_info = new_shard_conn.execute_command("search.clusterinfo")
-        if cluster_info[:2] == ['num_partitions', int(initial_shards_count+1)]:
-            break
-
+    with TimeLimit(40, "fail to acknowledge topology"):
```

Comment:
IMO we can rely on the default timeout. if the test times out we will know for sure something is wrong

Reply:
It's on purpose, we saw that 40 seconds is sufficient to determine that something is wrong...